### PR TITLE
Triton Server with IPEX backend example

### PR DESCRIPTION
**Pull request by mzyczyns**
These adds example of Triton Server usage with the IPEX backend. 
The provided example demonstrates the inference of Bert-Base and Bert-Large models with IPEX optimizations, supporting both fp32 and bf16, as well as introducing dynamic batching support.

### CODE REVIEW

File: examples/cpu/serving/triton/model_utils/config.pbtxt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    dims: [-1 ]
+  }
+  
+  ]
+instance_group [
+  {
+    count: 1
+    kind: KIND_CPU
+  }
+]
```
Comment (jgong5) [inline examples/cpu/serving/triton/model_utils/config.pbtxt:26]: Do we support multi-instances? If so, how are the CPU cores allocated to each instance and how are the threads affinitized?

File: examples/cpu/serving/triton/model_utils/bert_common/1/model.py
Comment (jgong5) [inline examples/cpu/serving/triton/model_utils/bert_common/1/model.py:1]: Do we have an example on LLM?

File: examples/cpu/serving/triton/model_utils/config.pbtxt (hunk 1 above)
Comment (mzyczyns) [reply examples/cpu/serving/triton/model_utils/config.pbtxt:26]: To run this example on multiple instances, users need to manually initiate the Docker container on each host instance using './start.sh host.' Additionally, for a distributed setup, it's recommended to incorporate a load balancer between the client and multiple hosts. As for now that's not the part of the example. The CPU core allocation and thread affinitization are handled within the './start.sh' script, specifically in lines 42-43.

File: examples/cpu/serving/triton/model_utils/bert_common/1/model.py
Comment (mzyczyns) [reply examples/cpu/serving/triton/model_utils/bert_common/1/model.py:1]: As of now, only examples for Bert-Base and Large inference have been included.

File: examples/cpu/serving/triton/model_utils/config.pbtxt (hunk 1 above)
Comment (jgong5) [reply examples/cpu/serving/triton/model_utils/config.pbtxt:26]: Can we add how to run multiple instances in the README doc?

File: examples/cpu/serving/triton/model_utils/bert_common/1/model.py
Comment (jgong5) [reply examples/cpu/serving/triton/model_utils/bert_common/1/model.py:1]: Is the support of LLM as simple as adding another model.py file?

File: examples/cpu/serving/triton/model_utils/config.pbtxt (hunk 1 above)
Comment (mzyczyns) [reply examples/cpu/serving/triton/model_utils/config.pbtxt:26]: As previously mentioned, incorporating a load balancer between the client and host is necessary, but it hasn't been implemented yet. The README provides instructions on how to run it in a supported scenario: on two separate instances (one for the client and one for the host).

File: examples/cpu/serving/triton/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Serving BERT models with Triton Server and Intel® Extension for PyTorch optimizations
+
+## Description
+This sample provide code to integrate Intel® Extension for PyTorch (IPEX) with Triton Inference Server framework. This project provides custom python backend for IPEX and additional dynamic batching algorithm to improve the performance. This code can be used as performance benchmark for Bert-Base and Bert-Large models.
```
Comment (jingxu10) [inline examples/cpu/serving/triton/README.md:4]: Intel® Extension for PyTorch (IPEX) -> Intel® Extension for PyTorch*
Please avoid using the abbreviation `IPEX` in doc, always use `Intel® Extension for PyTorch*`.

**Review by jingxu10 — CHANGES_REQUESTED** [review]

File: examples/cpu/serving/triton/README.md (hunk 2 above)
Comment (mzyczyns) [reply examples/cpu/serving/triton/README.md]: Removed IPEX abbreviation in README.md.

**Review by mzyczyns — COMMENTED** [review]
Changes added.

**Comment by jingxu10** [comment]
merged internally

**Pull request closed without merging**